src/virtualenv/seed/embed/via_app_data/pip_install/base.py

```diff
@@ -150,6 +147,36 @@ def _create_console_entry_point(self, name, value, to_folder, version_info):
         result.extend(Path(i) for i in new_files)
         return result
 
+    def _uninstall_previous_version(self):
+        dist_name = self._dist_info.stem.split("-")[0]
+        in_folders = chain.from_iterable([i.iterdir() for i in {self._creator.purelib, self._creator.platlib}])
```

Comment:
The installation happens only into the `purelib`: https://github.com/pypa/virtualenv/pull/2186/files#diff-05dbe4fcecab074226aa3cf0e1d6285e9a47e4a1c529a20ab49f2b572c3d49eaR38

Why bother with `platlib` when `install` will never actually touch it?

Reply:
If the package is either in the purelib or platlib is an implementation detail of the installer. Our installer uses purelib, but is completely legal for someone else to use platlib for it.

Reply:
Then it's even more important to restore the logic above. If the package being removed is in the `platlib` but the extraneous files are in `purelib` then while the package is removed from `platlib` the `purelib` will stay polluted and installation will fail.